### PR TITLE
Netcode shit

### PR DESCRIPTION
**Pull request by LJSonik**
I removed a line and the netcode was fixed.

-Fixed broken net commands, thus fixing (or at least greatly improving) chat/commands/joins/leaves and possibly other annoying bugs
-Updated packet name list so the debug file no longer shows garbage packet names
-Replaced byte values with actual net command names in the debug file. Only the first net command in a packet will be shown though
-Added a MOBJCONSISTANCY define that makes the game takes all revelant mobjs to be counted in the synch seed
-Added a PACKETDROP define that adds two console commands "drop" and "droprate" to simulate bad internet by dropping packets
-Added/changed comments here in there in the netcode
-Fixed a minor error that would ignore one of the urgent ack slots
-Added a space between the map name and "zone" for the messages shown in a joiner's console

### CODE REVIEW

File: src/d_clisrv.c [hunk 1]
```diff
@@ -400,8 +406,6 @@ static void ExtraDataTicker(void)
 				}
 			}
 		}
-
-	D_FreeTextcmd(gametic);
```
Comment (toasterone) [inline src/d_clisrv.c]: this is the line removal that improved the reliability, as pointed out on irc

File: src/d_net.c [hunk 2]
```diff
@@ -227,10 +232,13 @@ static boolean GetFreeAcknum(UINT8 *freeack, boolean lowtimer)
...
-			numfreeslote++;
-			if (netbuffer->packettype >= PT_CANFAIL && numfreeslote < URGENTFREESLOTENUM)
-				continue;
+			// For low priority packets, make sure to let freeslots so urgent packets can be sent
+			if (netbuffer->packettype >= PT_CANFAIL)
+			{
+				numfreeslot++;
+				if (numfreeslot <= URGENTFREESLOTNUM)
+					continue;
+			}
```
Comment (MonsterIestyn) [inline src/d_net.c]: Question: Does making numfreeslot only count PT_CANFAIL packettypes affect much?

**Comment by MonsterIestyn** [comment]
Tbh I'm still not sure if this should be merged to master or not, but that's just me. Otherwise all seems good at a glance, as far as I can tell.

**Comment by toasterone** [comment]
Honestly MI, the reason we have the Master/Next split is netgame compatibility. Considering it has no affect whilst you're a client and actively makes things better when you're the host, I think it's fair to say this is the most netgame-compatible codebase SRB2's had in years!

It looks good. I hesitate to plusone at this moment simply because I don't quite know what's happening with the packet memory now it's not being freed in the wrong place. Is it getting freed a few tics later?  Is it just forming a memory leak? My inability to sightread this and lack of knowledge about tools that could empirically stresstest this make me apprehensive.

I gues Alam is going to have to break the Valgrind out, maybe.

**Comment by LJSonik** [comment]
I made a test and it happens that the net command list is not correctly freed on client side, thus leaking memory.
It shouldn't be very hard to fix though.

**Comment by toasterone** [comment]
On... client side? But the mistaken free happens on server side!

**Comment by toasterone** [comment]
:+1:

Having seen data from testing locally, I can tell that the memory leak was never a problem for servers, and -
1) has been fixed in this branch now
2) was fine with the fix, because it was exactly what was going on when you connected with a 2.1.16a exe to a netgame hosted by someone using an exe based on this branch, which a ton of people did yesterday without running into any major problems.

Congratulations, @LJSonik.

**Comment by SteelTitanium** [comment]
Can confirm, everything checks out fine. :+1:

**Pull request merged**